src/plugins/data/public/ui/dataset_selector/_dataset_selector.scss

```
@@ -30,4 +30,20 @@
     flex: 1;
     align-self: center;
   }
+
+  &__button--truncate {
+    display: flex;
+    align-items: center;
+    max-width: 400px;
```

```suggestion
    max-width: 300px;
```
400px is too long for that bar on smaller screens

---

Fixed.